scripts/gen-deps-list

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
+const DepTree = require('deptree')
+const semver = require('semver')
+const { getPackages } = require('./utils')
+const [, , rootPackageName, incrementLevel] = process.argv
+
+if (!['major', 'minor', 'patch'].includes(incrementLevel)) {
+  console.error('Usage: gen-deps-list <package-name> <increment-level>')
+  console.error('  package-name: package name to generate dependencies list for')
+  console.error('  increment-level: semver increment level to use (major, minor, patch)')
+  process.exit(1)
+}
+
+class Package {
+  constructor({ name, version, dependencies = {}, optionalDependencies = {}, peerDependencies = {} }) {
+    this.name = name
+    this.version = version
+    this.requiredDependencies = dependencies
+    this.optionalDependencies = optionalDependencies
+    this.peerDependencies = peerDependencies
+  }
+
+  hasRequiredDependency(name) {
+    return this.requiredDependencies.hasOwnProperty(name)
+  }
+
+  hasOptionalDependency(name) {
+    return this.optionalDependencies.hasOwnProperty(name)
+  }
+
+  hasPeerDependency(name) {
+    return this.peerDependencies.hasOwnProperty(name)
+  }
+
+  /**
+   * @param {Package} dependency
+   * @param {string} incrementLevel
+   * @returns {boolean}
+   */
+  shouldHandleDependency(dependency, incrementLevel) {
+    if (this.hasRequiredDependency(dependency.name) || this.hasOptionalDependency(dependency.name)) {
+      return true
+    }
+
+    if (!this.hasPeerDependency(dependency.name)) {
+      return false
+    }
```

Comment:
Upgrading to a new incompatible peer dependency probably requires the package to be a breaking change too.